jack-core/src/rb/schema_rb_parser.rb

```diff
@@ -63,14 +63,14 @@ class Index
 
   class Table
     include FromHash
-    attr_accessor :name, :force, :id, :schema
+    attr_accessor :name, :force, :id, :limit, :options, :schema
     fattr(:columns) { [] }
 
     def __column(type, name, ops = {})
-      self.columns << Column.new(ops.merge(type: type, name: name)) unless FORBIDDEN_FIELD_NAMES.include?(name)
+      self.columns << Column.new(ops.merge(type: type, name: name)) unless FORBIDDEN_FIELD_NAMES.include?(name) || type == 'index'
```

Comment:
```suggestion
      if type == 'index'
        self.schema.add_index(self.name, name, ops)
      else
        self.columns << Column.new(ops.merge(type: type, name: name)) unless FORBIDDEN_FIELD_NAMES.include?(name)
      end
```

I tried to see if this would change the codegen result, but it's still the same output. I confirmed also by adding a `puts columns.each_with_index.count` / `puts columns.each_with_index.inspect` to [Table.to_model_defn](https://github.com/LiveRamp/jack/pull/287/files#diff-03619601ecc187e248019194c732b157a92700cce8ad37be82051ebf900bca99R83).